airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/check/CheckOperation.kt

```diff
@@ -13,13 +13,17 @@ import jakarta.inject.Singleton
 @Requires(property = Operation.PROPERTY, value = "check")
 @Requires(env = ["destination"])
 class CheckOperation(
+    // What is the purpose of this DestinationCheck interface?
+    // If CheckOperation is just a stub for now, then it shouldn't exist.
+    // If it's not just a stub, then its purpose is very much not clear to me, fwiw.
```

Comment:
1. Our philosophy is to expose simple interfaces at the implementor level, driven by more complex tasks at the CDK level.
2. So the Implementor contract for "Write" is to implement DestinationWrite; contract for "Check" is to implement `DestinationCheck`. There will be a default implementation that just syncs a single row to a test schema, but that will be an `@Seconday DefaultDestinationCheck` that can be overridden.
3. Especially, we don't want the implementor to have to specify a `@Requires` guard that refers to CDK internals (like Operation.PROPERTY).

Reply:
Oh got it. In my opinion, this philosophy is worth explaining in the javadoc comments, or other readers might be left with the same impression as I had.